source/extensions/network/dns_resolver/apple/apple_dns_impl.cc

```diff
@@ -347,6 +362,7 @@ void AppleDnsResolverImpl::PendingResolution::onDNSServiceGetAddrInfoReply(
     pending_response_.v4_responses_.clear();
     pending_response_.v6_responses_.clear();
 
+    addTrace(static_cast<uint8_t>(AppleDnsTrace::Failed));
```

Comment:
do we want to differentiate this failure mode?  AFIK it makes sense to

Reply:
There can be a lot of different possible failures and it's probably not easy to enumerate all of them. The [getaddrinfo](https://github.com/envoyproxy/envoy/blob/0e1e3d4b724e8d108815589e09df31132faed71e/source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc#L213) implementation also doesn't differentiate between different failures, except for the cases where we have special logic.